replica/table.cc

```diff
@@ -974,14 +978,38 @@ compaction_group& table::compaction_group_for_sstable(const sstables::shared_sst
     return _sg_manager->compaction_group_for_sstable(sst);
 }
 
-compaction_group_list& table::compaction_groups() const noexcept {
-    return _sg_manager->compaction_groups();
+// FIXME: not used in hot paths, but can be removed later to avoid allocation (using chunked to reduce reallocations).
+utils::chunked_vector<compaction_group*> table::compaction_groups() const {
```

Comment:
I think it would be better to introduce for_each_compaction_group helper rather than introducing a function + FIXME :)